src/core/algorithms/gfd/gfd_miner.cpp

```diff
@@ -0,0 +1,648 @@
+    std::vector<std::size_t> indices(boost::num_vertices(pattern));
+    std::iota(std::begin(indices), std::end(indices), 0);
+    std::vector<std::vector<std::size_t>> pairs = GetSubsets<std::size_t>(indices, 2);
+
+    std::vector<Literal> result = {};
+    for (auto const& index_pair : pairs) {
+        std::size_t fst = index_pair.at(0);
+        std::size_t snd = index_pair.at(1);
+        std::string fst_label = pattern[boost::vertex(fst, pattern)].attributes.at("label");
+        std::string snd_label = pattern[boost::vertex(snd, pattern)].attributes.at("label");
+
+        for (auto const& fst_name : attrs_info.at(fst_label)) {
+            Token fst_token = {fst, fst_name.first};
+            for (auto const& snd_name : attrs_info.at(snd_label)) {
+                Token snd_token = {snd, snd_name.first};
+                result.push_back({fst_token, std::move(snd_token)});
+            }
+        }
+    }
+    for (std::size_t i = 0; i < boost::num_vertices(pattern); ++i) {
+        std::string label = pattern[boost::vertex(i, pattern)].attributes.at("label");
+        for (auto& name : attrs_info.at(label)) {
+            Token name_token = {i, name.first};
+            for (auto& value : name.second) {
+                Token value_token = {-1, value};
+                Literal l = {name_token, std::move(value_token)};
+                result.push_back(std::move(l));
+            }
+        }
+    }
+    return result;
+}
+
+std::size_t Support(graph_t const& graph, Embeddings const& embeddings, graph_t const& pattern) {
+    std::map<vertex_t, std::set<vertex_t>> result = {};
+    BGL_FORALL_VERTICES_T(v, pattern, graph_t) {
+        for (Embedding const& embedding : embeddings) {
+            std::set<vertex_t> vertices = {embedding.at(v)};
+            auto [it, emplaced] = result.try_emplace(v, std::move(vertices));
+            if (!emplaced) {
+                it->second.insert(embedding.at(v));
+            }
+        }
+    }
+
+    std::size_t min_size = boost::num_vertices(graph);
+    for (auto const& maps : result) {
+        min_size = std::min(maps.second.size(), min_size);
+    }
+    return min_size;
+}
+
+void FilterSupp(graph_t const& graph, std::vector<graph_t>& patterns,
+                std::vector<Embeddings>& embeddings_set, std::size_t const sigma) {
+    std::vector<std::size_t> del_indices = {};
+    for (std::size_t i = 0; i < patterns.size(); ++i) {
+        if (Support(graph, embeddings_set.at(i), patterns.at(i)) < sigma) {
+            del_indices.push_back(i);
+        }
+    }
+    std::reverse(del_indices.begin(), del_indices.end());
+    for (auto& index : del_indices) {
+        patterns[index] = patterns.back();
+        patterns.pop_back();
+
+        embeddings_set[index] = embeddings_set.back();
+        embeddings_set.pop_back();
+    }
+}
+
+void Initialize(graph_t const& graph, std::set<std::string>& vertex_labels,
```

Comment:
This method is used only once so it worth making it a method so you don't `graph` and work with a field `graph_`